### DISCUSSION TITLE
Moving to typescript

### DISCUSSION THREAD

**Discussion by bastienrobert** [Accudio/async-alpine#51] [General]
I'm thinking of opening a PR to move the code to typescript (or at least generate typings and put them in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/)).
I think there might be some limitations due to the new API, regarding the new methods added to the `Alpine` object.

I've already converted the codebase and generated the types, there's still a bit of work to do on the bundler, but I think it's a good start.
https://github.com/bastienrobert/async-alpine/tree/feat/move-to-typescript

**Comment by Accudio** [comment]
Hi @bastienrobert.

I appreciate the effort but I will not accept a PR to convert the project to Typescript. For various reasons I won't get into here, I do not personally use or want to use Typescript on this project.

That said, I believe it is possible for a library based on JS to ship types that give hints and make it more usable as a dependency within Typescript projects. It's something that's been on my backlog to look into for a while as I'm not sure on exact details, but I haven't got around to it.
I would be open to including type definitions for that purpose if it's something you wanted to add, but no Typescript compilation step.

Thanks!

**Reply by Accudio** [reply]
Actually I think what I was saying with shipping types is what you've suggested with:
> (or at least generate typings and put them in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/)).

Have I understood that correctly? Definitely happy with including/providing types for those that use Typescript.

**Reply by bastienrobert** [reply]
@Accudio I just pushed a PR here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71462/files, can you review it?

**Reply by Accudio** [reply]
@bastienrobert I appreciate that, I do not think it's of value for me to review it. I haven't worked with Typescript, and this is the first I've heard of DefinitelyTyped so I cannot provide a meaningful review compared to a maintainer.

Looks like it may be approved regardless which is great. Thank you!

**Comment by Accudio** [comment]
Closing this discussion off as I don't believe there's anything more to discuss here, but feel free to correct me if appropriate!
